test/blackbox-tests/test-cases/pkg/just-print-solver-env.t

```diff
@@ -0,0 +1,28 @@
+Print the solver env when no dune-workspace is present
+  $ dune pkg lock --just-print-solver-env
+  Solver environment for context default:
+  ((flags (with-doc with-test)) (sys ()))
+
+Add some build contexts with different environments
+  $ cat >dune-workspace <<EOF
+  > (lang dune 3.8)
+  > (context
+  >  (default
+  >   (name linux)
+  >   (lock dune.linux.lock)
+  >   (solver_env
+  >    (sys
+  >     (os linux)))))
+  > (context
+  >  (default
+  >   (name no-doc)
+  >   (lock dune.linux.lock)
+  >   (solver_env
+  >    (flags (:standard \ with-doc)))))
```

Comment:
A little weird that we write `:with-doc` in the dune-project file, but `with-doc` in the workspacefile

Reply:
That wouldn't work currently because this field is an ordered set and the ordered set language only allows keywords to start with `:` (`:standard`, `:include`, etc). I actually think it's a little weird that when using variables in dependency conditions in `dune-project` that the `:` prefix is necessary to distinguish them from strings (e.g. it's really easy to make the mistake of filtering a dependency on `(= os "linux")` (which means the same as `(= "os" "linux")`)). I appreciate that it's hard to change now but I'd much rather we disallow unquoted strings in `dune-project` dependency specifications and treat all bare words as variables. I guess for now we can say something like "the `:` prefix dereferences a variable into its contained value".

Reply:
> I appreciate that it's hard to change now but I'd much rather we disallow unquoted strings in dune-project dependency specifications and treat all bare words as variables.

Yeah, that would create a ton of inconsistency of how we handle dune files. For better or for worse, we treat `foo` and `"foo"` as equivalent in positions where both are allowed.

> That wouldn't work currently because this field is an ordered set and the ordered set language only allows keywords to start with : (:standard, :include, etc).

This isn't a fundamental limitation and we can lift it or introduce some explicit form for variables whenever there's ambiguity . E.g. `(var foo)`.

For now we can leave this as is, but this is a rather big inconsistency to expose to users.